code/robert/python/papajokes.py

```diff
@@ -0,0 +1,58 @@
+import json, requests
+
+while True:
+    search = input("Type a search term: ")
+    url_page = []
+    url = "https://icanhazdadjoke.com/search?term="
+    url_search = f"{url}{search}{url_page}"
+
+    response = requests.get(url_search, headers = {'Accept': "application/json"})
+
+    print(response)
+    #200!!!!!
+
+    data = json.loads(response.text)
+
+    #print(type(data['results']))
+
+    print(f"There are {data['total_jokes']} jokes with that keyword and {data['total_pages']} pages")
+
+    for items in data['results']:
+
+        pages = range(0,int(data['total_pages']))
+        total_jokes = range(0,int(data['total_jokes']))
+        jokenumber = int(input(f"Select a random number for your joke (0 - {data['total_jokes']}) or any letter for a new search: "))
```

Comment:
Here in the input statement in the terminal `select a random number between  0-10 or any letter for a new search' if there are 10 items in the search results. and I choose 10, the last joke in the list of jokes, I get an error `IndexError: list index out of range` I think you can probably figure out why.  I will hold off on merging it unless you decide not to fix it. Required parts have been completed.